### Introduction to Derivatives

**Definition:** Financial instruments whose value is derived from underlying assets.

**Types:** Options, futures, forwards, swaps.

**Purpose:** Hedging, speculation, and arbitrage.


### Valuation of Options

The Black-Scholes Model is a widely used model for option pricing.

#### Formula:

$$
C = S_0N(d_1) - Xe^{-rT}N(d_2)
$$

Where:

- $C$: Call option price
- $S_0$: Current stock price
- $X$: Strike price
- $T$: Time to maturity
- $r$: Risk-free rate
- $N$: Cumulative distribution function of the standard normal distribution

#### $d_1$ and $d_2$:

$$
d_1 = \frac{\ln(S_0/X) + \left(r + \frac{\sigma^2}{2}\right)T}{\sigma \sqrt{T}}
$$

$$
d_2 = d_1 - \sigma \sqrt{T}
$$


### Practical Example

**Scenario:** Calculate the price of a call option using the Black-Scholes model.

**Steps:**

1. Identify the required inputs (stock price, strike price, time to maturity, risk-free rate, volatility).
2. Apply the Black-Scholes formula.


In [1]:
import numpy as np
from scipy.stats import norm

# Example data
S0 = 100  # Current stock price
X = 105   # Strike price
T = 1     # Time to maturity (1 year)
r = 0.05  # Risk-free rate (5%)
sigma = 0.20  # Volatility (20%)

# Calculate d1 and d2
d1 = (np.log(S0 / X) + (r + 0.5 * sigma**2) * T) / (sigma * np.sqrt(T))
d2 = d1 - sigma * np.sqrt(T)

# Calculate call option price
call_price = S0 * norm.cdf(d1) - X * np.exp(-r * T) * norm.cdf(d2)
print(f"Call Option Price: ${call_price:.2f}")


Call Option Price: $8.02


### Explanation of the Result

The calculated call option price using the Black-Scholes model is **$8.02**. This price represents the cost of the option to buy the underlying asset (stock) at the strike price ($105) one year from now, given the current stock price ($100), the risk-free rate (5%), and the volatility (20%).

- **$S_0$**: The current stock price is $100.
- **$X$**: The strike price is $105.
- **$T$**: The time to maturity is 1 year.
- **$r$**: The risk-free rate is 5%.
- **$\sigma$**: The volatility of the stock is 20%.

The price of $8.02 indicates the premium an investor must pay today to have the right, but not the obligation, to purchase the stock at $105 in one year. The price is influenced by factors such as the stock's volatility and the time to expiration.
